### PR TITLE
add matcher for extraneous closing parenthasis

### PR DESCRIPTION
**Pull request by thctamm**
Solves issue [138](https://github.com/cs50/help50/issues/138)

### CODE REVIEW

File: helpers/clang.py [hunk 1]
```diff
@@ -249,6 +249,20 @@ def help(lines):
 
     # $ clang foo.c
     # /tmp/foo-1ce1b9.o: In function `main':
+    # foo.c:1:19: error: extraneous ')' before ';' [-Werror]
+    # #include <stdio.h>;
+    #                   ^
+    matches = re.search(r"^([^:]+):(\d+):(\d+): (?:warning|error): extraneous ')' before ';'", lines[0])
```
Comment (dmalan) [inline helpers/clang.py]: Do use this file's `match` function instead (see file's bottom), which handles much of this regex.

File: helpers/clang.py [hunk 2]
```diff
@@ -249,6 +249,20 @@ def help(lines):
 
     # $ clang foo.c
     # /tmp/foo-1ce1b9.o: In function `main':
+    # foo.c:1:19: error: extraneous ')' before ';' [-Werror]
+    # #include <stdio.h>;
```
Comment (dmalan) [inline helpers/clang.py]: Why's this `#include` here?

File: helpers/clang.py [hunk 3]
```diff
@@ -249,6 +249,20 @@ def help(lines):
...
+    #                   ^
+    matches = re.search(r"^([^:]+):(\d+):(\d+): (?:warning|error): extraneous ')' before ';'", lines[0])
+    if matches:
+        response = [
+            "You seem to have an error in `{}` on line {}.".format(matches.group(1), matches.group(2), matches.group(3),
+            "You seem to have an extra closing parenthesis on this line. Make sure that every closing parenthesis has a matching opening parenthesis and vice versa.",
+            "You most likely have an extra closing parenthesis that you should remove or you are missing an opening parenthesis that you should add in."
```
Comment (dmalan) [inline helpers/clang.py]: Bit verbose. Let's change to:

```
response = [
    "You seem to have an extra parenthesis on line {} of `{}`, just before the semicolon."
]
```

Comment (dmalan) [inline helpers/clang.py]: Just a ping!

**Comment by thctamm** [comment]
Rebasing made this PR messy. Making a new one

**Pull request closed without merging**